File: verbalregex/src/main/java/ch/rasc/verbalregex/VerbalRegex.java
package ch.rasc.verbalregex;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

import ru.lanwen.verbalregex.VerbalExpression;

public class VerbalRegex {

  public static void main(String[] args) {
    VerbalExpression regex = VerbalExpression.regex()
        .startOfLine()
        .capture().range("A", "Z").count(2, 3).endCapture()
        .then("-")
        .capture().digit().count(1, 3).endCapture()
        .then(".")
        .capture().anyOf("xzy").endCapture()
        .endOfLine()
        .build();

    System.out.println(regex.toString());

    String[] inputs = new String[] { "AB-0.z", "ABC-99.y", "BB-789.x", "ab-999.x" };

    Pattern pattern = Pattern.compile(regex.toString());

    for (String input : inputs) {
      Matcher matcher = pattern.matcher(input);
      if (matcher.matches()) {
        System.out.printf("Group 1: %s, Group 2: %s, Group 3: %s%n", matcher.group(1),
            matcher.group(2), matcher.group(3));
      }
      else {
        System.out.println(input + " does not match");
      }
    }

  }

}

File: verbalregex/js/verbalregex.js
const VerEx = require('verbal-expressions');

const regex = VerEx()
                .startOfLine()
                .beginCapture()
                  .range("A", "Z").add("{2,3}")
                .endCapture()
                .then("-")
                .beginCapture()
                  .digit().add("{1,3}")
                .endCapture()
                .then(".")
                .beginCapture()
                  .anyOf("xyz")
                .endCapture()
                .endOfLine();

const inputs = ["AB-0.z", "ABC-99.y", "BB-789.x", "ab-999.x"];

for (const input of inputs) {
  regex.lastIndex = 0;
  const match = regex.exec(input);
  if (match != null) {
    console.log(`Group 1: ${match[1]}, Group 2: ${match[2]}, Group 3: ${match[3]}`);
  }
  else {
    console.log(`${input} does not match`);
  }
}
